core/src/main/java/org/infinispan/distribution/ch/impl/ReplicatedConsistentHashFactory.java

```diff
@@ -167,7 +167,7 @@ public Integer getId() {
 
       @Override
       public Set<Class<? extends ReplicatedConsistentHashFactory>> getTypeClasses() {
-         return Collections.<Class<? extends ReplicatedConsistentHashFactory>>singleton(ReplicatedConsistentHashFactory.class);
```

Comment:
Originally I had updated this class when converting to a protostream marshaller, then I reverted those changes but noticed that the type declaration wasn't required, so I updated that and changed to `Util.asSet` because that's what we have used in move of our Externalizers. I can change to just `Collections.singleton(..)` or  just not include this change if you prefer.

Reply:
I personally would prefer singleton, but looking at impl details of `Util.asSet` it actually uses singleton underneath if there was only a single entry. So I am fine either way.